e2e/integrations/integration.steps.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * Created by jludvice on 8.3.17.
+ */
+import { binding, then, when } from 'cucumber-tsflow';
+import { CallbackStepDefinition } from 'cucumber';
+import { expect, P, World } from '../common/world';
+import { IntegrationEditPage, ListActionsComponent } from '../integrations/edit/edit.po';
+import { log } from '../../src/app/logging';
+import { IntegrationsListPage } from '../integrations/list/list.po';
+
+/**
+ * Created by jludvice on 1.3.17.
+ */
+@binding([World])
+class IntegrationSteps {
+
+  constructor(protected world: World) {
+  }
+
+  @when(/defines integration name "([^"]*)"$/)
+  public defineIntegrationName(integrationName: string, callback: CallbackStepDefinition): void {
+    // Write code here that turns the phrase above into concrete actions
+    const page = new IntegrationEditPage();
+
+    page.basicsComponent().setName(integrationName).then(() => callback());
+  }
+
+
+  @then(/^she is presented with a visual integration editor$/)
+  public editorOpened(): P<any> {
+    // Write code here that turns the phrase above into concrete actions
+    const page = new IntegrationEditPage();
+    return expect(page.rootElement().isPresent(), 'there must be edit page root element')
+      .to.eventually.be.true;
+  }
+
+  @then(/^she is presented with a visual integration editor for "([^"]*)"$/)
+  public editorOpenedFor(integrationName: string): P<any> {
+    return this.editorOpened().then(() => {
+      // ensure we're on editor page and then check integration name
+        const page = new IntegrationEditPage();
+        return expect(page.flowViewComponent().getIntegrationName(), `editor must display integration name ${integrationName}`)
+          .to.eventually.be.equal(integrationName);
+      }
+    ).catch(e => P.reject(e));
+  }
```

Comment:
@jludvice more the trailing bracket together with curly bracket, it did the trick for me.

```
     })
.catch(e => P.reject(e));
// or
    }).catch(e => P.reject(e));

```